publication-commons/src/test/java/no/unit/nva/publication/service/impl/ResourceServiceTest.java

```diff
@@ -832,6 +842,88 @@ void shouldLogUserInformationQueryObjectAndResourceIdentifierWhenFailingToPublis
         assertThat(exception.getMessage(), containsString(RESOURCE_WITHOUT_MAIN_TITLE_ERROR));
     }
     
+    @Test
+    void shouldSaveDraftPublicationWithAssociatedFilesAsUnpublished() {
+        var publication = draftPublicationWithoutDoiAndAllTypesOfFiles();
+        var persisted = resourceService.createPublication(UserInstance.fromPublication(publication), publication);
+        assertThatAllFilesAreUnpublished(persisted);
+    }
+    
+    private static void assertThatAllFilesAreUnpublished(Publication persisted) {
+        for (var artifact : persisted.getAssociatedArtifacts()) {
+            assertThat(artifact, is(instanceOf(UnpublishedFile.class)));
+        }
+    }
+    
+    private Publication draftPublicationWithoutDoiAndAllTypesOfFiles() {
+        var legacyFile = randomFile().buildLegacyFile();
+        var publishedFile = randomFile().buildPublishedFile();
+        var unpublishedFile = randomFile().buildUnpublishedFile();
+        return draftPublicationWithoutDoi()
+                   .withAssociatedArtifacts(List.of(legacyFile, publishedFile, unpublishedFile))
+                   .build();
+    }
+    
+    @Test
+
+    void shouldSaveAlreadyPublishedPublicationWithAssociatedFilesAsPublished() throws NotFoundException {
+        var legacyFile = randomFile().buildLegacyFile();
+        var publishedFile = randomFile().buildPublishedFile();
+        var unpublishedFile = randomFile().buildUnpublishedFile();
+        var publication = randomPublication()
+                              .copy()
+                              .withStatus(PUBLISHED)
+                              .withAssociatedArtifacts(List.of(legacyFile, publishedFile, unpublishedFile))
+                              .build();
+        var persisted = resourceService.createPublicationFromImportedEntry(publication);
+        
+        assertThatAllAssosicatedArtifactsArePublished(persisted);
+    }
+    
+    private void assertThatAllAssosicatedArtifactsArePublished(Publication publication) throws NotFoundException {
+        var fetched = resourceService.getPublication(publication);
+        for (var artifact : fetched.getAssociatedArtifacts()) {
+            assertThat(artifact, is(instanceOf(PublishedFile.class)));
+        }
+    }
+    
+    @Test
```

Comment:
There is a test missing regarding AssociatedArtifacts that are not Files. It will be added on subsequent PR